do/serverless.go

```diff
@@ -680,52 +704,148 @@ func (s *serverlessService) GetConnectedAPIHost() (string, error) {
 	return s.owClient.Config.Host, nil
 }
 
-// ReadProject takes the path where project lies and validates the project.yml.
-// once project.yml is validated it reads the directory for all the files and sub-directory
-// and returns the struct of the files
-func (s *serverlessService) ReadProject(project *ServerlessProject, args []string) (ServerlessOutput, error) {
-	err := readTopLevel(project)
+// ReadProject takes the path where project lies and validates the project.yml and also
+// finds all the files and sub-directories of the project.  The information is merged
+// and returned as a ProjectSpec.
+func (s *serverlessService) ReadProject(project *ServerlessProject) (ProjectSpec, error) {
+	var (
+		empty  ProjectSpec
+		answer ProjectSpec
+	)
+	err := readProjectStructure(project)
 	if err != nil {
-		return ServerlessOutput{}, err
+		return empty, err
+	}
+	if project.ConfigPath != "" {
```

Comment:
Note that not all projects have a `project.yml`.   Some are expressed purely through file structure.   The (as yet unimplemented) merge of the two sources of information needs to deal correctly with that.